src/cmd/manage_nsfs.js

```diff
@@ -600,40 +599,78 @@ async function manage_account_operations(action, data, show_secrets, argv) {
+ * @param {string[]} [filters]
  */
-function filter_account_results(accounts, argv) {
-    //supported filters for list
-    const filters = _.pick(argv, ['uid', 'gid']); //if we add filters we should remove this comment and the comment 4 lines below (else)
-    return accounts.filter(item => {
-        for (const [key, val] of Object.entries(filters)) {
-            if (key === 'uid' || key === 'gid') {
-                if (item.nsfs_account_config && item.nsfs_account_config[key] !== val) {
-                    return false;
-                }
-            } else if (item[key] !== val) { // We will never reach here if we will not add an appropriate field to the filter 
+function filter_account(account, filters) {
+    for (const [key, val] of Object.entries(filters)) {
+        if (key === 'uid' || key === 'gid') {
+            if (account.nsfs_account_config && account.nsfs_account_config[key] !== val) {
                 return false;
             }
+        } else if (key === 'user') {
+            if (account.nsfs_account_config && account.nsfs_account_config.distinguished_name !== val) {
+                return false;
+            }
+        } else if (key === 'access_key') {
+            if (account.access_keys && account.access_keys[0][key] !== val) {
+                return false;
+            }
+        } else if (account[key] !== val) { // We will never reach here if we will not add an appropriate field to the filter 
+            return false;
         }
-        return true;
-    });
+    }
+    return true;
 }
 
+/**
+ * filter_bucket will return true or false based on whether a bucket meets the criteria defined by the supported flags.
+ * currently not implemented
+ * @param {object} bucket
+ * @param {string[]} [filters]
+ */
+function filter_bucket(bucket, filters) {
+    for (const [key, val] of Object.entries(filters)) {
+        if (bucket[key] !== val) { // We will never reach here if we will not add an appropriate field to the filter 
+            return false;
+        }
+    }
+    return true;
+}
 /**
  * list_config_files will list all the config files (json) in a given config directory
  * @param {string} config_path
+ * @param {boolean} [wide]
+ * @param {boolean} [show_secrets]
+ * @param {object} [filters]
  */
-async function list_config_files(config_path, show_secrets) {
+async function list_config_files(type, config_path, wide, show_secrets, filters) {
     const fs_context = native_fs_utils.get_process_fs_context();
     const entries = await nb_native().fs.readdir(fs_context, config_path);
+    const should_filter = Object.keys(filters).length > 0;
 
     let config_files_list = await P.map_with_concurrency(10, entries, async entry => {
         if (entry.name.endsWith('.json')) {
-            const full_path = path.join(config_path, entry.name);
-            const data = await get_config_data(full_path, show_secrets);
-            return data;
+            if (wide || should_filter) {
```

Comment:
@romayalon please add these changes to CLI doc,

Reply:
what do you mean? we currently don't put every option of the CLI in the docs but I think we should in a different PR that takes everything from help and describes everything more deeply
I added it to the help for now